### PR TITLE
TravisCI changes

### PR DESCRIPTION
**Pull request by mgerhardy**
- always include the demo windows in compilation to check for errors
- compile the example
- use -Wall

### CODE REVIEW

**Comment by mgerhardy** [comment]
See warnings shown in https://travis-ci.org/vurtun/nuklear/jobs/323619677

**Comment by mgerhardy** [comment]
- What do you think about using -Werror for TravisCI? In my opinion a header only library should be free from warnings. So a CI should check for them.
- What do you think about activating appveyor? I could provide a CI config for it.

**Comment by mgerhardy** [comment]
Rebased against latest master.

File: demo/glfw_opengl3/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 BIN = demo
 
 # Flags
-CFLAGS = -std=c99 -pedantic -O2
+CFLAGS ?= -std=c99 -pedantic -O2
```
Comment (dumblob) [inline demo/glfw_opengl3/Makefile]: Well, in this case I would probably prefer `CFLAGS += -std=c99 -pedantic -O2`. What do you think?

Comment (mgerhardy) [reply demo/glfw_opengl3/Makefile]: Changed. Also all other makefiles are changed now.

**Comment by vurtun** [comment]
>- always include the demo windows in compilation to check for errors

Sure why not.

>- compile the example

Sure.

>- use -Wall

I compile with:

```
# Modes
.PHONY: clang
clang: CFLAGS += -Weverything -Wno-padded -Wno-missing-noreturn -Wno-covered-switch-default -Wno-float-equal -Wno-comma -Wthread-safety
clang: CFLAGS += -Wno-double-promotion -Wno-switch-enum -Wno-format-nonliteral
clang: CC = clang
clang: $(BIN)

.PHONY: gcc
gcc: CFLAGS += -Wall -Wextra -Wformat-security -Wunreachable-code -Wlogical-op
gcc: CFLAGS += -Winline -Wshadow -Wwrite-strings -Wdeprecated-declarations
gcc: CFLAGS += -Wstrict-prototypes -Wold-style-definition -Wconversion
gcc: CFLAGS += -Wredundant-decls -Wnested-externs -Wmissing-include-dirs
gcc: CFLAGS += -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wmissing-prototypes
gcc: CFLAGS += -Wswitch-default -Wundef -Wno-unused -Wstrict-overflow=5 -Wsign-conversion
gcc: CFLAGS += -Winit-self -Wstrict-aliasing -Wconversion -Wshift-count-negative -Wshift-count-overflow
gcc: CFLAGS += -Wswitch-enum -Winvalid-pch -Wbad-function-cast
gcc: CFLAGS += -Wno-switch-enum
gcc: CC = gcc
gcc: $(BIN)

```
I don't want these flags inside the provided Makefile. However if you are talking about CTravisCI than yeah no  problem having -Wall specified.

>- What do you think about using -Werror for TravisCI? In my opinion a header only library should be free from warnings. So a CI should check for them.

-Werror is a bad idea. Problem are different compiler version generating different warnings.

>- What do you think about activating appveyor? I could provide a CI config for it.

Never worked with it but sure why not.

**Pull request merged**